jctools-core/src/main/java/org/jctools/queues/MpscArrayQueue.java

```diff
@@ -562,4 +565,17 @@ public void fill(Supplier<E> s, WaitStrategy w, ExitCondition exit)
             idleCounter = 0;
         }
     }
+    
+    public List<E> unorderedSnapshot() {
+        int length = capacity();
+        List<E> elements = new ArrayList<E>();
+        for (int i = 0; i < length; i++) {
```

Comment:
You iterate through the whole array, very likely to be many thousands of elements, to find the elements you need

Reply:
I'll try to make it less wasteful.